bundles/org.openhab.binding.magentatv/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,238 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="magentatv"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="receiver">
+		<label>MagentaTV Media Receiver</label>
+		<description>Represents a Telekom Media Receiver for MagentaTV</description>
+
+		<channel-groups>
+			<channel-group id="control" typeId="control">
+				<label>Control</label>
+			</channel-group>
+			<channel-group id="program" typeId="program">
+				<label>Program Information</label>
+			</channel-group>
+			<channel-group id="status" typeId="status">
+				<label>Play Status</label>
+			</channel-group>
+		</channel-groups>
+
+
+		<config-description>
+			<parameter name="udn" type="text">
+				<label>Unique Device Name</label>
+				<description>The UDN identifies the Media Receiver</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="modelId" type="text">
+				<label>Receiver Model</label>
+				<description>Model type (MR4xx/ MR2xx)</description>
+				<required>true</required>
+				<options>
+					<option value="Auto">AUTO</option>
+					<option value="DMS_TPB">MR400</option>
+					<option value="MR401B">MR401B</option>
+					<option value="MR201">MR201</option>
+				</options>
+			</parameter>
+			<parameter name="ipAddress" type="text">
+				<label>Media Receiver's IP Address</label>
+				<description>IP Address of the receiver</description>
+				<required>true</required>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="text">
+				<label>Receiver Port</label>
+				<description>Port address for UPnP</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="accountName" type="text">
+				<label>Account Name</label>
+				<description>Credentials: Login name (xxx@t-online.de, same as for the Telekom Kundencenter)</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="accountPassword" type="text">
+				<label>Account Password</label>
+				<description>Credentials: Account Password (same as for the Telekom Kundencenter)</description>
+				<context>password</context>
+				<required>true</required>
```

Comment:
No, this is not required. Only in case it is there, it will be used to automatically retrieve the userID.

Reply:
Should be made an advanced parameter.

Reply:
no, the standard procedure is to use the credentials

Reply:
No, the standard way is to provide mandatory parameters, everything else is a workaround.
We can think about finding solutions in openHAB 3 for your kind of problem - what you need is a kind of better support for setting up things, but you should not mistake this for what is required for technically executing the handler on a day-by-day basis.

To make me better understand the situation: Can you explain how exactly the userId that is retrieved from the account is mapped to the Thing? E.g. if you find two receivers via UPnP in the local network: Is there a way to figure out to which the userId belongs to (or rather: Can there be potentially be two MagentaTV subscriptions in the account, which would result in two different userIds)?

Reply:
ok, we have 1 account with 2 receivers
- all receivers need to be signed up for that account (using your credentials)
- the binding discovers them using UPnP - 1,2..n, same account, different accounts, whatever
- The pairing process requires a userID, this is some kind of technical ID attached to the account
- from our tests it seems that a given credential always creates the same userID( maybe some kind of salted hash)
- This userID might be known (at the beginning we used Wireshark with the App and could use the discovered ID)
- or is a result of performing the OAuth login with the credentials.
- Therefore you could enter the userID in the thing config or the login credentials and the binding performs OAuth and saves the userID to the thing config

The existing solution might not be perfect, but it's convenient for the user.
- discovering the thing, adding from the Inbox and things shows OFFLINE is confusing for the user, because he doesn't know that the credentials are required (who reads the manual...)
- using a console command might be an alternative, but various users are even not aware of the OH console or how to use it (I saw that more than once in the community threads)

Having a proper setup process in OH 3.0 is a good idea. This scenario here is not special. I see various other bindings requiring and storing textual credentials (e.g. Fritz, Fritz AHA, Fritz!Box, Gardena and others) to get access to a device or a cloud account etc. Some kind of interaction between the binding and the user would be perfect
a) giving users hints on configuration while discovering or adding them
b) prompt for data like credentials during the inclusion process or
c) displaying expressive information while something failed
Those messages could be displayed to the user when he's using the UI or queued in a kind of notification log.

and by the way: You mentioned the console command of the Tesla bidning, but it exactly does the same: this is optional, otherwise you could setup the credentials in the UI - where is the difference?
From the Doc:
```
If you do not want to use the openHAB console, you can also manually create a "Tesla Account" thing in the UI and provide your username and password as parameters...
```

Nevertheless, how we proceed here? The current solution is well received by the users and doesn't create issues. I could add a console command so users have the choice of entering the credentials directly into the thing config or triggering the process from the console, which then just saves the userID to the thing config. This is one tool mnore to use during setup, which creates a challenge to beginners.

Reply:
Thanks for the elaborate explanation.

> I could add a console command so users have the choice of entering the credentials directly into the thing config or triggering the process from the console

Yes, this is exactly my proposal, to have two possible ways here:
1. get hold of the UID by providing username+password in the config - this option should only be used when creating the Thing through the UI, since in this case the binding has a chance to wipe the credentials after retrieving the UID.
2. get hold of the UID by a console command - this option is very useful when textually configuring the Thing, since users would not want to put their credentials in that file and thus have a way to find out the UID for putting it into the Thing file.

But in both situations the running system won't find any credentials within the Thing configuration -> therefore, these fields are "not required" and must be marked just as that.

> Having a proper setup process in OH 3.0 is a good idea.

Fully agree, but let's keep that out of scope here - feel free to create an issue in openhab-core where we can discuss this (probably best with @ghys being in the loop as well).

Reply:
Mechanics fully refactored, see comment below